sklearn/ensemble/tests/test_iforest.py

```diff
@@ -165,19 +165,19 @@ def test_iforest_parallel_regression():
     assert_array_almost_equal(y1, y3)
 
 
-def test_iforest_performance():
+def test_iforest_performance(global_random_seed):
     """Test Isolation Forest performs well"""
 
     # Generate train/test data
-    rng = check_random_state(2)
-    X = 0.3 * rng.randn(120, 2)
+    rng = check_random_state(global_random_seed)
+    X = 0.3 * rng.randn(1200, 2)
     X_train = np.r_[X + 2, X - 2]
-    X_train = X[:100]
+    X_train = X[:1000]
 
     # Generate some abnormal novel observations
-    X_outliers = rng.uniform(low=-4, high=4, size=(20, 2))
-    X_test = np.r_[X[100:], X_outliers]
-    y_test = np.array([0] * 20 + [1] * 20)
+    X_outliers = rng.uniform(low=-16, high=16, size=(200, 2))
+    X_test = np.r_[X[1000:], X_outliers]
+    y_test = np.array([0] * 200 + [1] * 200)
```

Comment:
This test is already not tested what it's supposed to.
```py
X = 0.3 * rng.randn(120, 2)  # generate 120 points centered at the origin
X_train = np.r_[X + 2, X - 2]  # duplicate such that 120 points centered at (-2,-2) and 120 at (2,2)
X_train = X[:100]  # ignore the line just above and take 100 points centered at the origin
```
Then same for X_test...

The test is also brittle. When testing multiple seeds it often happen that the X_test taken from X are not mixed with the ones of X_train and thus are misclassified.
I propose to modify the test as follows, with a little bit less samples than you proposed
```py
    # Generate train/test data
    rng = np.random.RandomState(global_random_seed)
    X = 0.3 * rng.randn(600, 2)
    X = rng.permutation(np.vstack((X + 2, X - 2)))
    X_train = X[:1000]

    # Generate some abnormal novel observations
    X_outliers = rng.uniform(low=-1, high=1, size=(200, 2))
    X_test = np.vstack((X[1000:], X_outliers))
    y_test = np.array([0] * 200 + [1] * 200)
```

Reply:
Hi @jeremiedbb, I've updated the test case as you suggested, and verifies that it works for all the random seeds :)